widgets/cdatetime/org.eclipse.nebula.widgets.cdatetime/src/org/eclipse/nebula/widgets/cdatetime/CDateTime.java

```diff
@@ -1168,990 +1196,1023 @@ void handleKey(Event event) {
+	 * its contents are attempted to be committed. If the commit is successful,
+	 * focus switches to the next field of CDateTime. Otherwise, the contents of
+	 * <code>this.editField</code> are discarded and the previous value of the
+	 * selection (before user started typing in this field) is restored; focus
+	 * stays in the same field. <b>Example:</b> if the seconds field contains
+	 * "23", and user types 8 in the seconds field, "08" is shown on screen
+	 * while getSelection still returns 23. If user types 9 after that, the
+	 * field reaches its capacity, the attempt to commit 89 seconds fails, and
+	 * 23 gets restored on screen.
+	 * 
+	 * @param e
+	 *            the event
+	 * @see CDateTime#updateText()
+	 */
+	void verify(Event e) {
+		// we don't want to reprocess the event
+		if (e.doit == false) {
+			return;
+		}
+
+		e.doit = false;
+		if (field.length == 0 || activeField == FIELD_NONE) {
+			return;
+		}
+
+		char c = e.character;
+		if (e.text.length() == 1 && String.valueOf(c).equals(e.text)
+				&& Character.isDigit(c) || e.text.length() > 1) {
+			if (e.text.length() == 1) {
+				if (editField == null) {
+					int cf = getCalendarField();
+					if (cf >= 0) {
+						int digits;
+						switch (cf) {
+						case Calendar.YEAR:
+							digits = 4;
+							break;
+						case Calendar.DAY_OF_YEAR:
+							digits = 3;
+							break;
+						case Calendar.AM_PM:
+						case Calendar.DAY_OF_WEEK:
+						case Calendar.ERA:
+							digits = 1;
+							break;
+						case Calendar.MILLISECOND:
+							digits = 3;
+							break;
+						default:
+							digits = 2;
+						}
+						editField = new EditField(digits, calendar.get(cf));
+					} else {
+						return;
+					}
+				}
+				if (editField.addChar(c)) {
+					if (commitEditField()) {
+						fieldNext();
+					} else {
+						editField = null;
+						if (selection.length > 0) {
+							selection[0] = calendar.getTime();
+						}
+						updateText();
+					}
+				}
+				if (selection.length > 0) {
+					selection[0] = calendar.getTime();
+				}
+				updatePicker();
+			} else {
+				try {
+					setSelection(df.parse(e.text));
+					fireSelectionChanged();
+				} catch (ParseException pe) {
+					// do nothing
+				}
+			}
+		}
+		updateText();
+	}
+
+	/**
+	 * @param pattern
+	 * @param allowedTimeZones
+	 * @throws IllegalArgumentException
+	 */
+	public void setPattern(final String pattern,
+			final TimeZone[] allowedTimeZones) throws IllegalArgumentException {
+		this.setPattern(pattern);
+		if (pattern.indexOf('z') != -1) {
+			this.allowedTimezones = allowedTimeZones;
+		}
+	}
+	
+	/** 
+	 * Returns the minimum date or <code>null</code>.
+	 * 
+	 * @return Returns the minDate.
+	 * @since 1.4.0
+	 */
+	public Calendar getMinDate() {
+		return minDate;
+	}
+
+	/** 
+	 * Returns the maximum date or <code>null</code>.
+	 * 
+	 * @return Returns the maxDate.
+	 * @since 1.4.0
+	 */
+	public Calendar getMaxDate() {
```

Comment:
Allows maxdate to be altered. Is this your intention? If so then state that in the javadoc. If not then return a clone of maxdate.